### PR TITLE
docs: remove toc from first tool guide and add unique ids to markdown headers

### PR DESCRIPTION
**Pull request by njhale**
- The default table of contents is incompatible with tabs, so remove it
  from the first-tool guide
- Auto-generated IDs for markdown headers in the first-tool page aren't
  unique, so generate unique IDs and inject them at snippet fetch time
- Re-fetch latest README.md files from `*-hash-tool` repos to fix
  guide doc formatting issues

### CODE REVIEW

File: docs/snippets/python-hash-tool-readme.md.mdx [hunk 1]
```diff
@@ -78,16 +82,20 @@ Param: algo: The algorithm to generate a hash with. Default is "sha256". Support
...
 The `tool.gpt` file also provides the following metadata for use in `Otto8`:
 
 - `!metadata:*:category` which tags Tools with the `Crypto` category to promote organization and discovery
 - `!metadata:*:icon` which assigns `https://cdn.jsdelivr.net/npm/@phosphor-icons/core@2/assets/duotone/fingerprint-duotone.svg` as the Tool icon
 
-Where `*` is a wild card pattern that applies the metadata to all Tools in a `tool.gpt`.
+<br/>
 
-```text
+> **Note:** `*` is a wild card pattern that applies the metadata to all Tools in a `tool.gpt`.
```
Comment (thedadams) [inline docs/snippets/python-hash-tool-readme.md.mdx]: There is a "callout" for things like this.
```suggestion
:::note

`*` is a wild card pattern that applies the metadata to all Tools in a `tool.gpt`.

:::
```

File: docs/snippets/python-hash-tool-readme.md.mdx [hunk 2]
```diff
@@ -276,36 +296,70 @@ To do this, run through the following steps in the root of your local fork:
...
-   ![Enter Tool Repo Reference](https://raw.githubusercontent.com/otto8-ai/python-hash-tool/refs/heads/main/docs/add-tools-step-2.png "Enter Tool Repo Reference")
-
-Afterwords, the Tool will be available for use in `Otto8`.
-
-You can search for the Tool by category or name on the Tools page to verify:
-
-![Search For Newly Added Tools](https://raw.githubusercontent.com/otto8-ai/python-hash-tool/refs/heads/main/docs/add-tools-step-3.png "Search For Newly Added Tools")
-
-### Using The `Hash` Tool in an Agent
+1. <details>
```
Comment (thedadams) [inline docs/snippets/python-hash-tool-readme.md.mdx]: Curious why we need HTML instead of Markdown?

File: docs/src/plugins/fetch-snippets/index.ts [hunk 3]
```diff
@@ -1,16 +1,15 @@
 import fs from 'fs';
 import path from 'path';
 import axios from 'axios';
+import crypto from 'crypto';
 import { LoadContext, Plugin } from '@docusaurus/types';
 
-// Array of permalinks to raw files in different repositories
 const FILE_URLS = [
-  'https://raw.githubusercontent.com/otto8-ai/python-hash-tool/main/README.md',
+  'https://raw.githubusercontent.com/njhale/python-hash-tool/main/README.md',
```
Comment (thedadams) [inline docs/src/plugins/fetch-snippets/index.ts]: Is this an expected change?

File: docs/src/plugins/fetch-snippets/index.ts [hunk 4]
```diff
@@ -62,6 +64,25 @@ async function fetchFiles(outputDir: string) {
   );
 }
 
+// Function to add unique IDs to Markdown headers
+function addUniqueHeaderIds(url, markdown: string): string {
```
Comment (thedadams) [inline docs/src/plugins/fetch-snippets/index.ts:68]: Why is this necessary? Can we just add the IDs directly?

File: docs/src/plugins/fetch-snippets/index.ts (hunk 3 above)
Comment (njhale) [reply docs/src/plugins/fetch-snippets/index.ts]: nope! this was leftover from testing, thank you for catching that.

File: docs/snippets/python-hash-tool-readme.md.mdx (hunk 2 above)
Comment (njhale) [reply docs/snippets/python-hash-tool-readme.md.mdx]: HTML gives us better control the size and position of the images in this document. In this case I'm also using `<details>` to keep them hidden until the user clicks on the step, greatly reducing visual clutter.

If there's a better way to do this without HTML, I'm all ears.

File: docs/snippets/python-hash-tool-readme.md.mdx (hunk 1 above)
Comment (njhale) [reply docs/snippets/python-hash-tool-readme.md.mdx]: Is this valid markdown, or is it `mdx`? These files are automatically fetched at build time from the example repos; in this case https://github.com/otto8-ai/python-hash-tool/blob/main/README.md, so it's nice to stick to GitHub compatible markdown if we can.

File: docs/src/plugins/fetch-snippets/index.ts (hunk 4 above)
Comment (njhale) [reply docs/src/plugins/fetch-snippets/index.ts:68]: This makes it easier for folks to fetch and use Markdown from any repo together without having to make changes to those repos to make the IDs unique.

Also, the syntax to add these IDs is `mdx`, so any IDs would render in GitHub's markdown preview:

Screenshot 2024-11-14 at 1 17 22 PM

**Review by thedadams — APPROVED** [review]
